avalon/maya/pipeline.py

```diff
@@ -447,12 +448,44 @@ def parse_container(container, validate=True):
 
 
 def _ls():
-    containers = list()
-    for identifier in (AVALON_CONTAINER_ID,
-                       "pyblish.mindbender.container"):
-        containers += lib.lsattr("id", identifier)
+    """Yields Avalon container node names.
 
-    return containers
+    Used by `ls()` to retrieve the nodes and then query the full container's
+    data.
+
+    Yields:
+        str: Avalon container node name (objectSet)
+
+    """
+
+    def _maya_iterate(iterator):
+        """Helper to iterate a maya iterator"""
+        while not iterator.isDone():
+            yield iterator.thisNode()
+            iterator.next()
+
+    ids = {AVALON_CONTAINER_ID,
+           # Backwards compatibility
+           "pyblish.mindbender.container"}
+
+    # Iterate over all 'set' nodes in the scene to detect whether
+    # they have the avalon container ".id" attribute.
+    fn_dep = om.MFnDependencyNode()
+    iterator = om.MItDependencyNodes(om.MFn.kSet)
+    for mobject in _maya_iterate(iterator):
+        if mobject.apiTypeStr != "kSet":
```

Comment:
Was this necessary? I would have thought the initial `MFn.kSet` to perform filtering?

Reply:
That was my assumption too but it seems it also gave nodes inhereted from set type, like shading engines if I tecall correctly. Something with materials at least.

Reply:
Tested this once more. It does indeed also return "shadingEngine" nodes which are an inherited type of the objectSet. As such this if statement helps in optimzing slightly. :)